mqtt/mqtt-broker/tests/translation.rs

```diff
@@ -0,0 +1,264 @@
+use matches::assert_matches;
+use mqtt3::{
+    proto::{ClientId, QoS},
+    ReceivedPublication,
+};
+
+use common::{TestClient, TestClientBuilder};
+use mqtt_broker::{AuthId, BrokerBuilder};
+
+mod common;
+
+// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#retrieving-a-device-twins-properties
+#[tokio::test]
+async fn translation_twin_retrieve() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let mut server_handle = common::start_server(broker);
+
+    let mut edge_hub_core = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("edge_hub_core".into()))
+        .build();
+    let mut device_1 = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("device_1".into()))
+        .build();
+
+    // Core subscribes
+    edge_hub_core
+        .subscribe("$edgehub/+/twin/get/#", QoS::AtLeastOnce)
+        .await;
+
+    // device requests twin update
+    device_1
+        .subscribe("$iothub/twin/res/#", QoS::AtLeastOnce)
+        .await;
+    device_1
+        .publish_qos1("$iothub/twin/GET/?rid=10", "", false)
+        .await;
+
+    // Core recieves request
+    recieve_with_topic(&mut edge_hub_core, "$edgehub/device_1/twin/get/?rid=10").await;
```

Comment:
I think its harder to read, since the assert ends up split onto 5 or 6 lines. I find it easier to read with the function, especially since the same assert is used like 20 times